server/api/crud/pipelines.py

```diff
@@ -262,6 +270,12 @@ def create_pipeline(
 
         logger.debug("Writing pipeline to temp file", content_type=content_type)
 
+        env_vars = server.api.api.utils.create_secret_env_vars_for_pipeline(auth_info)
```

Comment:
make sure you delete the secret when pipeline is deleted

Reply:
@liranbg - you cannot do that. The auth secret relates to a given username/access-key combination. It's very likely that the same auth-secret will be used by multiple runtimes/workflows, and it should only be deleted once the access-key is deleted (though we don't really do that today...)

Reply:
@theSaarco @liran Couldn't we use an K8S owner annotation to GC the secret once the workflow runner pod is deleted?